src/wallet/wallet.cpp

```diff
@@ -93,7 +93,7 @@ static void ReleaseWallet(CWallet* wallet)
     // so that it's in sync with the current chainstate.
     wallet->WalletLogPrintf("Releasing wallet\n");
     wallet->BlockUntilSyncedToCurrentChain();
-    wallet->Flush();
+    wallet->Flush(true);
```

Comment:
This is not right, this will mess other dbs in the same env.

Reply:
Thanks @promag . Fixed